llvm/lib/YkIR/YkIRWriter.cpp

```diff
@@ -40,13 +130,92 @@ class YkIRWriter {
     OutStreamer.emitInt8(0); // null terminator.
   }
 
+  void serialiseOpcode(OpCode Code) { OutStreamer.emitInt8(Code); }
+
+  void serialiseConstantOperand(Instruction *Parent, llvm::Constant *C) {
+    OutStreamer.emitInt8(OperandKind::Constant);
+    OutStreamer.emitSizeT(constantIndex(C));
+  }
+
+  void serialiseLocalVariableOperand(Instruction *I) {
+    // For now we assume that there is a one to one relationship between LLVM
+    // instructions and Yk IR instructions, and that the instruction
+    // (and block) indices are the same in both IRs.
+    BasicBlock *ParentBlock = I->getParent();
+    Function *ParentFunc = ParentBlock->getParent();
+
+    size_t BlockIdx = getIndex(ParentFunc, ParentBlock);
+    size_t InstIdx = getIndex(ParentBlock, I);
+
+    OutStreamer.emitInt8(OperandKind::LocalVariable);
+    OutStreamer.emitSizeT(BlockIdx);
+    OutStreamer.emitSizeT(InstIdx);
+  }
+
+  void serialiseStringOperand(const char *S) {
+    OutStreamer.emitInt8(OperandKind::String);
+    serialiseString(S);
+  }
+
+  void serialiseUnimplementedOperand(Value *V) {
+    // FIXME: just chuck out a string for now
```

Comment:
`YKFIXME` ?

Reply:
Fixed in 498ed84d9c719357dbd5acfa201ec5ebec814e26